lib/services/directory-synchronizer.js

```diff
@@ -0,0 +1,144 @@
+import { getFiles } from '../util/file';
+import { NTriplesMutator } from './n-triples-mutator';
+import { query } from '../util/database';
+import moment from 'moment';
+import groupBy from 'lodash.groupby';
+import { SemanticFile } from '../entities/semantic-file';
+import { SERVICE_URI } from '../../env';
+import { VersionFile } from '../entities/version-file';
+
+/**
+ * Abstract service providing basic synchronization between files on-disk and in-store
+ */
+export class DirectorySynchronizer {
+
+  constructor(root) {
+    this.debug = false;
+    this.root = root;
+    this.publisher = `${SERVICE_URI}${root}`;
+    this.files = [];
+    this.mutator = new NTriplesMutator();
+  }
+
+  /**
+   * Synchronize the files saved in-store with the once saved on-disk.
+   *
+   * @returns {Promise<{missing, finished}>}
+   */
+  async sync() {
+    const onDisk = await this.getSemanticFilesOnDisk();
+    const inStore = await this.getSemanticFilesInStore();
+    let {finished, missing} = await this.doSync(onDisk, inStore);
```

Comment:
I was a bit confused with the naming of `finished`, would something like ... `processed` fit ?

Reply:
yes, that's better! I'll rename it!